### PR TITLE
Testing JavaScript Tic-tac-toe

### PR DESCRIPTION
**Pull request by evabanegacom**
Added tests for the following modules
- Gameplay
- Game-board module

### CODE REVIEW

File: tests/gameBoardModule.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { gameBoardModule } from '../main';
+
+describe('checking gameboard', () => {
+  const board = ['', '', '', '', '', '', '', '', ''];
+  const newBoard = ['', 'X', '', '', '', '', '', '', ''];
+
+  test('should check for winning combos', () => {
+    expect(gameBoardModule.checkWinCombo()).toBe(null);
+  });
```
Comment (Torres-ssf) [inline tests/gameBoardModule.test.js]: - [ ] We need to implement more tests for the checkWinCombo functionality. You only check for the negative case. We need to test if it's returning true when the board has any of the possible winning cases. ( horizontal, vertical, and diagonal). Meaning, test for horizontal winning cases, vertical winning cases and of course, diagonal winning cases,

- [ ] We also need to implement a test case for the Tie situation.

File: tests/gameBoardModule.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  test('check if the reset board is correct', () => {
+    gameBoardModule.boardArr[1] = 'X';
+    expect(gameBoardModule.boardArr).toEqual(newBoard);
+  });
+
+  test('check if the reset board is correct', () => {
+    gameBoardModule.boardArr[1] = 'X';
+    expect(gameBoardModule.boardArr).toContain('X');
+  });
```
Comment (Torres-ssf) [inline tests/gameBoardModule.test.js]: - [ ] I'm confused about how these 2 tests are checking for a reset board. Isn't the reset board functionality already been checked in the other test module? I think these 2 tests are not doing much, we can think in other things that we can test in your application, like if the player's names are been properly displayed in their turn. The ideal scenario is to test all the features of your application

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi team :hand: 
Nice job so far testing your Tic Tac Toe Game, which I enjoyed a lot by the way. Congratulations on your game :smile: 
But before moving on we need to implement more test cases.

- [ ] The render method was never been tested.

Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

File: tests/gameBoardModule.test.js (hunk 2 above)
Comment (BrittanyBlake) [reply tests/gameBoardModule.test.js]: My bad! Neither of these is meant to reset the board. These two were meant to check if the board is being rendered correctly after an item is placed on the board.

**Comment by BrittanyBlake** [comment]
As per TSE review: 

- We have implemented tests for all of the win combinations and for a tie.
- updated the description of the existing tests to show what is being tested

**Review by wathigo — APPROVED** [review]
### Approved :heavy_check_mark: 
Great job guys :clap: 
All the requirements have been met.

**You may move on.**

**Good Luck :four_leaf_clover: 
Simon Wathigo.**

**Pull request merged**